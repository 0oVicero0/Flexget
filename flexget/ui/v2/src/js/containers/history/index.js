import { connect } from 'react-redux';
import HistoryPage from 'components/history';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(HistoryPage);
