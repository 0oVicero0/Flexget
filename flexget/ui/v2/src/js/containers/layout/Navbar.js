import { connect } from 'react-redux';
import { logout } from 'actions/auth';
import Navbar from 'components/layout/Navbar';

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(logout()),
  };
}

export default connect(null, mapDispatchToProps)(Navbar);

