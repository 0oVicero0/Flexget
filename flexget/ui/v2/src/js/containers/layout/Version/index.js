import { connect } from 'react-redux';
import { GET_VERSION } from 'actions/version';
import { request } from 'utils/actions';
import Version from 'components/layout/Version';

export function mapStateToProps({ version }) {
  return {
    version,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getVersion: () => dispatch(request(GET_VERSION)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Version);
