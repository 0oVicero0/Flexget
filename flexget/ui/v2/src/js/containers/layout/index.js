import { connect } from 'react-redux';
import { checkLogin } from 'actions/auth';
import Layout from 'components/layout';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
