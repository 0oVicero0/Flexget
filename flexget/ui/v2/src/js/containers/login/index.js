import { connect } from 'react-redux';
import LoginPage from 'components/login';

export function mapStateToProps({ auth }) {
  return {
    redirectToReferrer: !!auth.loggedIn,
  };
}

export default connect(mapStateToProps)(LoginPage);
