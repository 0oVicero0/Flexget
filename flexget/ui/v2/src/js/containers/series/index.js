import { connect } from 'react-redux';
import {
  GET_SHOWS,
} from 'actions/series/shows';
import SeriesPage from 'components/series';
import { request } from 'utils/actions';

function mapStateToProps({ series }) {
  return {
    shows: series.shows.items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getShows: payload => dispatch(request(GET_SHOWS, payload)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SeriesPage);
