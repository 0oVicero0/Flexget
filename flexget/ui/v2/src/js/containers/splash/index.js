import { connect } from 'react-redux';
import Splash from 'components/splash';
import { LOGIN, checkLogin } from 'actions/auth';

export function mapStateToProps({ status }) {
  return {
    checking: !!status.loading[LOGIN],
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Splash);
