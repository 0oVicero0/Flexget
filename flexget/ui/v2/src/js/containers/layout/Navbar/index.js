import { connect } from 'react-redux';
import { LOGOUT } from 'actions/auth';
import { SERVER_RELOAD, SERVER_SHUTDOWN } from 'actions/server';
import { request } from 'utils/actions';
import Navbar from 'components/layout/Navbar';

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(request(LOGOUT)),
    reloadServer: () => dispatch(request(SERVER_RELOAD)),
    shutdownServer: () => dispatch(request(SERVER_SHUTDOWN)),
  };
}

export default connect(null, mapDispatchToProps)(Navbar);

