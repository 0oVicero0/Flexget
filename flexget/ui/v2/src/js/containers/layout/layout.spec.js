import { mapStateToProps } from 'containers/layout';

describe('containers/layout', () => {
  it('should return logged in if logged in', () => {
    expect(mapStateToProps({ auth: { loggedIn: true } })).toMatchSnapshot();
  });

  it('should return not logged in if logged out', () => {
    expect(mapStateToProps({ auth: { } })).toMatchSnapshot();
  });
});
