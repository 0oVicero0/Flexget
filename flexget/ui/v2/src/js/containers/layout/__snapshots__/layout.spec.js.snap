// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/layout should return logged in if logged in 1`] = `
Object {
  "loggedIn": true,
}
`;

exports[`containers/layout should return not logged in if logged out 1`] = `
Object {
  "loggedIn": false,
}
`;
